### PR TITLE
Pr

### PR DESCRIPTION
**Pull request by JeniaSavenko**

### CODE REVIEW

File: app/controllers/user.controller.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
+const jwt = require('jsonwebtoken');
+const UserModel = require('./../model/user.model');
+const NoteModel = require('./../model/note.model');
+const connect = require('../../config/db');
+const { ObjectID } = require('mongodb');
+const secretKey = '123456789';
+const expiresIn = '1h';
```
Comment (Alex-Vasilev) [inline app/controllers/user.controller.js]: to config file

File: app/controllers/user.controller.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  app.post('/auth/reg', (req, res) => {
+    const { name, password } = req.body;
+    UserModel.find()
+      .then((notes) => {
+        if (isAuthenticated({
+          name,
+          password,
+          notes,
+        }) === true) {
+          const status = 401;
```
Comment (Hardy1207) [inline app/controllers/user.controller.js]: I think status codes should be in separate file.
`const UNAUTHORIZED_STATUS_CODE = 401`
and use like this -> ` res.status(UNAUTHORIZED_STATUS_CODE)`

File: app/controllers/user.controller.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        // Save Note in the database
+        user.save()
+          .then((data) => {
+            res.send(data);
+          });
+      })
+      .catch((err) => {
+        res.status(500)
+          .send({
+            message: err.message || 'Some error occurred while retrieving notes.',
```
Comment (Hardy1207) [inline app/controllers/user.controller.js]: Error strings should be in separate file

File: package.json [hunk 4]
```diff
@@ -19,6 +19,7 @@
   "homepage": "https://github.com/JeniaSavenko/nodeJsServer#readme",
   "dependencies": {
     "axios": "0.18.0",
+    "bcrypt": "^3.0.6",
```
Comment (Hardy1207) [inline package.json]: incremental dependency **22** and **38** `line`

File: server.js [hunk 5]
```diff
@@ -1,17 +1,43 @@
...
 const io = require('socket.io');
+const connect = require('./config/db');
+const Model = require('./app/model/user.model');
 
 const port = 3000;
+const secretKey = '123456789';
 const Request = require('./app/controllers/note.controller');
 
+
+const verifyToken = token => jwt.verify(token, secretKey, (err, decode) => (decode !== undefined ? decode : err));
```
Comment (Hardy1207) [inline server.js]: If **null** is invalid value for you too, use `decode ? decode : err;`

File: app/controllers/user.controller.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+function isAuthenticated({ name, password, notes }) {
+  return notes.findIndex(user => user.name === name && user.password === password) !== -1;
+}
+
+module.exports = (app) => {
+  app.post('/auth/login', (req, res) => {
+    const { name, password } = req.body;
+    UserModel.find()
+      .then((notes) => {
+        if (isAuthenticated({
```
Comment (Hardy1207) [inline app/controllers/user.controller.js]:
```
if (isAuthenticated({
          name,
          password,
          notes,
        }) === false)
```
Maybe it should looks like 
```
if (!isAuthenticated({
          name,
          password,
          notes,
        }))
```

File: app/controllers/user.controller.js [hunk 7]
```diff
@@ -1,28 +1,90 @@
...
+const createToken = payload => jwt.sign(payload, secretKey, { expiresIn });
+
+function isAuthenticated({ name, password, notes }) {
+  return notes.findIndex(user => user.name === name && user.password === password) !== -1;
+}
+
+module.exports = (app) => {
+  app.post('/auth/login', (req, res) => {
+    const { name, password } = req.body;
+    UserModel.find()
```
Comment (Alex-Vasilev) [inline app/controllers/user.controller.js]: how are you looking for current user?

**Pull request merged**